packages/x-data-grid/src/hooks/features/columns/useGridColumns.tsx

```diff
@@ -96,6 +96,7 @@ export function useGridColumns(
 
       apiRef.current.setState(mergeColumnsState(columnsState));
       apiRef.current.publishEvent('columnsChange', columnsState.orderedFields);
+      apiRef.current.updateRenderContext?.();
```

Comment:
It's defined in `useGridVirtualScroller` which isn't a top-level hook, it's a hook used in `GridVirtualScroller`. So it doesn't get initialized first. Not sure why `useGridVirtualScroller` was implemented that way. And yes this is called before the function is defined sometimes, but in those cases it doesn't matter that it's not called because it's during the 1st render, and the render context is updated right after the 1st render.

This should probably be refactored to move the hook where it should be, but I haven't had a reason to do the refactor other than it would be cleaner.